addons/hr_expense/models/hr_expense.py

```diff
@@ -747,6 +748,7 @@ def _prepare_payments_vals(self):
             'tax_tag_ids': to_update['tax_tag_ids'],
             'amount_currency': amount_currency,
             'currency_id': self.currency_id.id,
+            'partner_id': self.vendor_id.id,
```

Comment:
```suggestion
            'partner_id': self.vendor_id.id if self.payment_mode == 'company_account' else False,
```
Shouldn't `partner_id` be `False` if the expense is not paid by the company (several occurrences)?

Reply:
own_account do not go through this method (as discussed)